### ISSUE TITLE
Reconsider Licensing

### ISSUE THREAD

**Issue by Havvy** [mozilla/nixpkgs-mozilla#17]
> I also think this repository should be using MIT instead of the MPL, such that we can at least contribute back to Nixpkgs. -- @nbp in #15.

**Comment by Havvy** [comment]
Right now mozilla/nixpkgs-mozilla is MPL licensed. As this is Mozilla code, there's not a problem with that. But nixpkgs code is traditionally MIT and Rust code is traditionally MIT/Apache dual-licensed. If we want this code to somehow benefit nixpkgs in the future, we'll want it licensed under MIT, and if we want it possibly benefiting Rustup in the future (should they decide to support Nix) we'll want to also have Apache license.

I think the easiest way right now would be to triple license under all three.

**Comment by zimbatm** [comment]
Given that MIT is the most permissive license it's enough to just use MIT I think. I don't think it's possible to re-license Apache to MIT because of the patent restrictions but the other way around should be fine.

**Comment by garbas** [comment]
💯 I guess I selected MPL just out of habit for every Mozilla project 😄

**Comment by zimbatm** [comment]
Just in case, I agree to license all my contributions under the MIT

**Comment by garbas** [comment]
@zimbatm sorry I should have asked.

I understood your comment above as 👍 to a change. I hope I also understood @nbp's comment as thumbs up for license change.

**Comment by zimbatm** [comment]
tldr; yeah no worries, my comment is intended for lawyers :D

I always assume my contributions are licensed under the same terms as the project's license. I believe this should be the industry standard instead of making contributors sign CLAs. The project owners (or anybody else) can re-license the code under a more restrictive version without asking me as it's allowed by the original license. If the project wants to re-license to a more permissive license then the authors need my approval as I didn't transfer my contribution's copyright. Anyways, that's the long-winded reasoning.
